src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -72,7 +75,7 @@ public void selectPreviousBestMatchingMilestone() {
     }
 
     private boolean canIncreaseIndex(Optional<Integer> curSelectedIndex, List<PickerMilestone> list) {
-        return curSelectedIndex.isPresent() && curSelectedIndex.get() < list.size() - 1;
+        return !curSelectedIndex.isPresent() || curSelectedIndex.get() < list.size() - 1;
```

Comment:
We can use `filter` like: `return curSelectedIndex.filter((index) -> index < list.size() - 1).isPresent()`. `flatMap` can also be used in similar way.

But you can keep it as it is if you think the current way is more readable.